Added some features

EliasKotlyar opened the pull request:
Hello,

I included the functionality trough the "ath=1" GET-parameter

Please merge!

Greetings
Elias

jonathanhodges commented:
@EliasKotlyar So, I like this feature and I want to include it but it needs to check the application mode, development vs. production, unless you can demonstrate that is happening somewhere else.  Basically, being able to turn this on from the frontend, represents a security risk on a production server since it exposes information about the server on the frontend of the website.  So if you add a check to ensure it is in development mode before enabling the display of the template hints, that should be sufficient.

Thanks.

EliasKotlyar commented:
Hello , @jonathanhodges 

i have refactored the code,and added a check for Development IP. Also this new Version includes a working Remote-Call Functionality.

Please have a review and merge it. Many Thanks in Advance,

Greetings
Elias

jonathanhodges commented:
I have reviewed this and I don't see any specific issues. I will note I have not tested what you changed but I'm going to go ahead and merge it in.
